src/zope/testrunner/tests/testrunner-arguments.rst

```diff
@@ -40,28 +40,28 @@ different default, that one takes effect.
     >>> from zope import testrunner
     >>> testrunner.run_internal(defaults, 'test --layer 111'.split())
     Listing samplelayers.Layer111 tests:
-      test_x1 (sample1.sampletests.test111.TestA)
-      test_y0 (sample1.sampletests.test111.TestA)
-      test_z0 (sample1.sampletests.test111.TestA)
-      test_x0 (sample1.sampletests.test111.TestB)
-      test_y1 (sample1.sampletests.test111.TestB)
-      test_z0 (sample1.sampletests.test111.TestB)
-      test_1 (sample1.sampletests.test111.TestNotMuch)
-      test_2 (sample1.sampletests.test111.TestNotMuch)
-      test_3 (sample1.sampletests.test111.TestNotMuch)
+      test_x1 (sample1.sampletests.test111.TestA...)
+      test_y0 (sample1.sampletests.test111.TestA...)
+      test_z0 (sample1.sampletests.test111.TestA...)
+      test_x0 (sample1.sampletests.test111.TestB...)
+      test_y1 (sample1.sampletests.test111.TestB...)
+      test_z0 (sample1.sampletests.test111.TestB...)
+      test_1 (sample1.sampletests.test111.TestNotMuch...)
+      test_2 (sample1.sampletests.test111.TestNotMuch...)
+      test_3 (sample1.sampletests.test111.TestNotMuch...)
```

Comment:
I don't understand this change.

The commit says it's needed to make things compatible with Python 3.11.  What does Python 3.11 produce for the test names?

Oh, it repeats the test name inside the parentheses, like

> test_something (sampletests_buffering.TestSomething1.test_something)

Couldn't we have fixed that with a renormalizer?  Adding a bazillion `...`s to all the test files seems like a pain.  Of course that's a past pain now, and changing everything back to use a renormalizer would be even more pain.

Reply:
I was thinking about a renormalizer but did not find a good solution: The test names are too different to be normalized easily. And doctests do not yet seem to repeat the test name in the output.